src/main/java/com/savvato/tribeapp/services/UserServiceImpl.java

```diff
@@ -25,6 +28,13 @@ public class UserServiceImpl implements UserService {
 	@Autowired
 	SMSChallengeCodeService smsccs;
 
+	@Override
+	public Long getLoggedInUserId(){
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		UserDetails userDetails = (UserDetails) authentication.getPrincipal();
```

Comment:
I think here, you can just use the UserPrincipal that comes from `authentication.getPrincipal()`, and call `.getId()` on it. See the class UserPrincipal.java, and UserDetailsServiceTRIBEAPP.java. The TRIBEAPP class returns a UserDetails object to Spring, but it is actually a UserPrincipal object, which Tribe App defines, and when this UserPrincipal is created, it is initialized with a User object, which you should be able to use to get the user ID. I believe this will work. and it simplifies your code a bit.

Reply:
refactored

Reply:
Actually, just use the return from `authentication.getPrincipal()`. No need to do findByName(), etc..  You should be able to do `return authentication.getPrincipal().getId();`

Reply:
okay. It works when you cast it as UserPrincipal:
```
public Long getLoggedInUserId(){
		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
		return ((UserPrincipal) authentication.getPrincipal()).getId();
	}
```